src/haz3lweb/view/Deco.re

```diff
@@ -115,7 +115,8 @@ module Deco =
         };
       let range: option((Measured.Point.t, Measured.Point.t)) = {
         // if (Piece.has_ends(p)) {
-        switch (TermRanges.find_opt(Piece.id(p), M.term_ranges)) {
+        let id = Id.Map.find(Piece.id(p), M.terms) |> Term.rep_id;
```

Comment:
> the decoration when the cursor is on a comma in a partial application are missing the f (it works correctly when on the parentheses):
> ![image](https://github.com/hazelgrove/hazel/assets/108375845/aded2285-dda7-4b0f-9c68-a608569ab5e6)

The issue I encountered while fixing this bug is that the `TermRanges` seems to be related to `Skel`. Even if I can break down and reconstruct `Term` in `MakeTerm`, I'm not sure how to do the same for the corresponding `Skel`. It appears that the creation of `Skel` doesn't take into account the information from `Term`.

Here is the solution that Haoxiang came up with. I'm not sure if it will have any impact on existing programs, but it appears to be the least disruptive solution to the codebase.

Reply:
Interesting.... TermRanges are currently not working quite right in dev for both list literals and case rules, so it's not the end of the world if it doesn't work here. I don't really know the details of TermRanges so @dm0n3y will need to weigh in on whether this change is likely to be disruptive or not

Reply:
looks good to me. in fact, probably should have been written this way originally (using the rep_id).